SAT/Z3Interfacing.cpp

```diff
@@ -98,50 +98,63 @@ const char* errToString(Z3_error_code code)
   }
 }
 
-void handleZ3Error(Z3_context ctxt, Z3_error_code code) 
+void handleZ3Error(Z3_context ctxt, Z3_error_code code)
 {
   DEBUG(errToString(code))
   throw z3::exception(errToString(code));
 }
 
 #define STATEMENTS_TO_EXPRESSION(...) [&]() { __VA_ARGS__; return 0; }()
 
-Z3Interfacing::Z3Interfacing(SAT2FO& s2f, bool showZ3, bool unsatCore):
-  _varCnt(0), 
+Z3Interfacing::Z3Interfacing(SAT2FO& s2f, bool showZ3, bool unsatCore, vstring const& exportSmtlib):
+  _varCnt(0),
   _sat2fo(s2f),
-  _status(SATISFIABLE), 
+  _status(SATISFIABLE),
   _config(),
   _context(_config),
   _solver(_context),
   _model((STATEMENTS_TO_EXPRESSION(
-            BYPASSING_ALLOCATOR; 
-            _solver.check(); 
+            BYPASSING_ALLOCATOR;
+            _solver.check();
           ),
-         _solver.get_model())), 
+         _solver.get_model())),
   _assumptions(),
   _showZ3(showZ3),
-  _unsatCore(unsatCore)
+  _unsatCore(unsatCore),
+  _out(exportSmtlib == "" ? Option<std::ofstream>()
```

Comment:
Interesting. This seems like it needs to bypass the allocator. We have a bypass allocator call in the body, so does that happen before the initialisation? I'm not sure of the current allocator bypass checks.

Reply:
Oh good spot. I just checked and it only create instances of `Z3Interfacing` in contexts where we call `BYPASSING_ALLOCATOR` anyways. I will fix this in order to be future proof anyway.